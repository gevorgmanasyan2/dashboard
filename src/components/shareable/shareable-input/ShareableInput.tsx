import React, { FC } from 'react'
import { Input } from 'antd'
import './shareableInput.scss'
import { SharableProps } from '../models'

const ShareableInput: FC<SharableProps> = ({
  label,
  placeholder,
  value,
  name,
  onChange,
  className,
  onBlur,
  errorsMessage,
  readOnly
}) => {
  return (
    <div className="input__container">
      <label htmlFor="">{label}</label>
      <Input
        className={`input__container--input ${className}`}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
        name={name}
        onBlur={onBlur}
        readOnly={readOnly}
      />
      {errorsMessage && <div className='input__container--error'>{errorsMessage}</div>}
    </div>
  )
}

export default ShareableInput
