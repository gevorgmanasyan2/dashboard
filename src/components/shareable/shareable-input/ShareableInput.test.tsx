/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import ShareableInput from './ShareableInput'

it('should render ShareableInput component', () => {
  const mockItem = {
    label: 'Serial',
    placeholder: 'Name',
    onChange: () => {},
    onBlur: () => {},
    errorsMessage: 'error',
    className: 'text',
    value: 'test',
    name: 'test',
  }
  expect(shallow(<ShareableInput {...mockItem} />)).toMatchSnapshot()
})
