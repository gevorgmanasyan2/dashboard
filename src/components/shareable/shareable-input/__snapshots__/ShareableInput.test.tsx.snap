// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShareableInput component 1`] = `
<div
  className="input__container"
>
  <label
    htmlFor=""
  >
    Serial
  </label>
  <Input
    className="input__container--input text"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value="test"
  />
  <div
    className="input__container--error"
  >
    error
  </div>
</div>
`;
